### DISCUSSION TITLE
Processing Larger Files

### DISCUSSION THREAD

**Discussion by Adam-Arrayen** [roboflow/supervision#1089] [Q&A] [answered]
My code works fine on smaller files; however, I have an issue when working with files above 900MB. 
Just output a 1 second video. That one second looks right with the line, bounding boxes and labels, but of course not processing the entire video. 

Is there a way to work with larger files? 
Using supervision.__version__: 0.19.0

```
import cv2
import supervision as sv
from ultralytics import YOLO
import numpy as np

model = YOLO("yolov8x.pt")
SOURCE_VIDEO_PATH = r"<Source Video>"
TARGET_VIDEO_PATH = f"<Output location>"

START = sv.Point(0, 1500)
END = sv.Point(3840, 1500)

line_zone = sv.LineZone(start=START, end=END)

line_zone_annotator = sv.LineZoneAnnotator(
    thickness=4,
    text_thickness=4,
    text_scale=2)

byte_tracker = sv.ByteTrack()

bounding_box_annotator = sv.BoundingBoxAnnotator(thickness=2)
label_annotator = sv.LabelAnnotator(text_thickness=2, text_scale=1)
trace_annotator = sv.TraceAnnotator(thickness=4)

# class_ids of interest - person, car
selected_classes = [0, 2]

def callback(frame: np.ndarray, index:int) -> np.ndarray:
    results = model(frame, verbose=False)[0]
    detections = sv.Detections.from_ultralytics(results)
    detections = detections[np.isin(detections.class_id, selected_classes)]
    detections = byte_tracker.update_with_detections(detections)

    labels = [
        f"{results.names[class_id]} {confidence:0.2f}"
        for class_id, confidence
        in zip(detections.class_id, detections.confidence)
    ]

    annotated_frame = frame.copy()
    annotated_frame = trace_annotator.annotate(
        scene=annotated_frame,
        detections=detections)
    annotated_frame = bounding_box_annotator.annotate(
        scene=annotated_frame,
        detections=detections)
    annotated_frame = label_annotator.annotate(
        scene=annotated_frame,
        detections=detections,
        labels=labels)

    line_zone.trigger(detections)

    return  line_zone_annotator.annotate(annotated_frame, line_counter=line_zone)

sv.process_video(
    source_path = SOURCE_VIDEO_PATH,
    target_path = TARGET_VIDEO_PATH,
    callback=callback
)
```

**Comment by LinasKo** [comment]
Hi @Adam-Arrayen :wave:

I've managed to reproduce a similar issue.
Running your code on a 1GB-sized video only processed about 1 sec of the video and crashed with this warning:
```
[ WARN:0@8.268] global cap_ffmpeg_impl.hpp:1541 grabFrame packet read max attempts exceeded, if your video have multiple streams (video, audio) try to increase attempt limit by setting environment variable OPENCV_FFMPEG_READ_ATTEMPTS (current value is 4096)
```

Can you try setting the env variable it suggests to a higher value?

For example, I've had success with:
```bash
# Scope the env var to a single command
OPENCV_FFMPEG_READ_ATTEMPTS=16384 python main.py
```

If you're in a Colab environment, the process is different, but let me know and I can help out ;)

**Reply by Adam-Arrayen** [reply]
Hi @LinasKo, 
Thanks for getting back to me. 

I'm just running the script above as a Jupiter notebook, so didn't see the error message until I made it a python script and got the same message.

 So where do I make that change to increase the OPENCV_FFMPEG_READ_ATTEMPTS ?

**Comment by LinasKo** [comment, marked as answer]
That's an environmental variable. Typically they're set outside the Python script. A few examples, but I recommend option 3:

1. If you're running in a Linux / Mac terminal for just one session, you can run:
```bash
OPENCV_FFMPEG_READ_ATTEMPTS=16384
python your_python_script_with_supervision.py
```

2. Or, as I did:
```bash
OPENCV_FFMPEG_READ_ATTEMPTS=16384 python your_python_script_with_supervision.py
```

3. A cleaner way is to use `.env` files.

Create a file that's just called `.env`, inside it only write:
```bash
OPENCV_FFMPEG_READ_ATTEMPTS=16384
```
(other env variables would be in seperate lines, no need for any commas at the end or anything)

Then, when you have an env file, you need to load it in python. `pip install python-dotenv`, and then in your script do:
```python
from dotenv import load_dotenv

load_dotenv()
```

This should make OpenCV see it, and it should work in both scripts and Jupyter notebooks.

4. There's ways to set it from within Python with `os`, but that's not common - it goes against the idea of env variables a little bit.

5. If you wanted to set it permanently for your system, that differs on where you are - in Linux it's `~/.bashrc` file, on Windows it's some digging through menus, on Mac - I'm not sure.

---

Update:

Tested in Colab, and it seems that at least some notebooks have a simpler way:

```
# from: https://stackoverflow.com/a/53311791/3369193
%env OPENCV_FFMPEG_READ_ATTEMPTS=16384

# Test
!echo $OPENCV_FFMPEG_READ_ATTEMPTS
```

**Reply by Adam-Arrayen** [reply]
Cheers, I couldn't get the dotenv .env file to work, but I will keep trying that solution. 
As i'm using Windows did below from the command line: 
```
set OPENCV_FFMPEG_READ_ATTEMPTS=16384
python my_python_script_with_supervision.py
```
and that worked with a larger video file. 

Thanks for your help @LinasKo

**Reply by LinasKo** [reply]
Good to hear :)

Best of luck in your project!
